File: 02_python/coin.py
# 동전0
import numpy as np

while True:
    try:
        coin_cnt = int(input("동전 종류의 개수를 입력해주세요: "))
        break
    except ValueError:
        print("숫자를 입력해 주세요")

while True:
    try:
        n = int(input("금액을 입력해주세요: "))
        break
    except ValueError:
        print("숫자를 입력해 주세요")

type_array = []

for i in range(coin_cnt):
    while True:
        try:
            coin_type = int(input("동전의 종류를 입력해주세요: "))
            break
        except ValueError:
            print("숫자를 입력해 주세요")
    type_array.append(coin_type)

type_array=np.sort(type_array)[::-1]
#print(type_array)
cnt = 0

for coin in type_array:
    cnt += n // coin
    n %= coin


print("최소 동전 개수: ", cnt)

File: 02_java/src/coin.java
// 거스름 돈: 최소한의 동전 개수
import java.util.Arrays;
import java.util.Collections;
import java.util.Scanner;

public class coin {

  public static void main(String[] args) {

    Scanner scan = new Scanner(System.in);
    System.out.print("동전 종류의 개수를 입력해주세요: ");
    int coin_cnt = scan.nextInt();

    System.out.print("금액을 입력해주세요: ");
    int n = scan.nextInt();

    Integer [] type_array = new Integer [coin_cnt];

    for (int i = 0; i < coin_cnt; i++){
      System.out.print("동전의 종류를 입력해주세요: ");
      type_array[i] = scan.nextInt();
    }
    Arrays.sort(type_array, Collections.reverseOrder());

    int cnt = 0;
    for (int k : type_array) {
      cnt += n / k;
      n %= k;
    }
    System.out.println("최소 동전 개수: "+cnt);
  }
}
